src/compiler/emitter.ts

```diff
@@ -55,6 +55,20 @@ module ts {
     }
 
     let indentStrings: string[] = ["", "    "];
+
+    // When emitting for ES3, property access via dot notation (i.e. obj.prop ) cannot be used
+    // if the property name is a reserved word.  Use index notation (i.e. obj["prop"]).
+    let ES3ReservedWords = [

```

Comment:
Based on our previous conversation, it sounds like we will need syntax kinds here.  That way we can report errors properly if you do things like write "function transient" and then try to emit for ES3.